LazyProxy/LazyProxyBuilder.cs

```diff
@@ -64,9 +69,30 @@ public static Type GetType(Type type)
         /// <returns>The lazy proxy type instance.</returns>
         public static T CreateInstance<T>(Func<T> valueFactory)
         {
-            var lazy = new Lazy<T>(valueFactory);
-            var proxyType = GetType<T>();
-            return (T) Activator.CreateInstance(proxyType, lazy);
+            return (T) CreateInstance(typeof(T), () => valueFactory());
```

Comment:
This does double wrapping of delegates, if T is reference type (it should be because it is an interface).
We could simply cast `Func<T>` to `Func<object>` because of `Func` is covariant.